flink-metrics/flink-metrics-influxdb/src/test/java/org/apache/flink/metrics/influxdb/MetricMapperTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.metrics.influxdb;
+
+import org.apache.flink.metrics.Counter;
+import org.apache.flink.metrics.Gauge;
+import org.apache.flink.metrics.Histogram;
+import org.apache.flink.metrics.HistogramStatistics;
+import org.apache.flink.metrics.Meter;
+
+import org.influxdb.dto.Point;
+import org.junit.Test;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test for {@link MetricMapper} checking that metrics are converted to InfluxDB client objects as expected.
+ */
+public class MetricMapperTest {
+
+	private final String name = "a-metric-name";
+	private final MeasurementInfo info = getMeasurementInfo(name);
+	private final Instant timestamp = Instant.now();
+
+	@Test
+	public void testMapGauge() {
+		verifyPoint(
+			MetricMapper.map(info, timestamp, (Gauge<Number>) () -> 42),
+			"value=42");
+
+		verifyPoint(
+			MetricMapper.map(info, timestamp, (Gauge<Number>) () -> null),
+			"value=null");
+
+		verifyPoint(
+			MetricMapper.map(info, timestamp, (Gauge<String>) () -> "hello"),
+			"value=hello");
+
+		verifyPoint(
+			MetricMapper.map(info, timestamp, (Gauge<Long>) () -> 42L),
+			"value=42");
+	}
+
+	@Test
+	public void testMapCounter() {
+		Counter counter = mock(Counter.class);
+		when(counter.getCount()).thenReturn(42L);
+
+		verifyPoint(
+			MetricMapper.map(info, timestamp, counter),
+			"count=42");
+	}
+
+	@Test
+	public void testMapHistogram() {
+		HistogramStatistics statistics = mock(HistogramStatistics.class);
```

Comment:
replace mock with actual implementation

Reply:
Are there any reasons to not use mocks in this test class?

Reply:
there's no reason to rely on another library when you can just implement the interface. you mock every method anyway.
Actually implementing the interface is more maintainable as methods added to the interface have to be taken care of in every test, instead of flying under the radar breaking somehow, sometime, somewhere down the line.

Reply:
`Mockito` is already used in many places of Flink codebase.
Only methods required for a test are mocked. With inheritance, all abstract methods need to be implemented (even if they are not related to a test), which is more verbose in Java and makes some noise.

> Actually implementing the interface is more maintainable as methods added to the interface have to be taken care of in every test, instead of flying under the radar breaking somehow, sometime, somewhere down the line.

It's not obvious for me, how using an interface in a test is more maintainable.
Mock also "implements" the interface. If the interface changes, the mock may also need to be updated.
If something changes, but the test passes and allows to sneak a bug. Then it's question of test improvement, rather if some details use mocks or real, imo.